tests/config/test_config_spec.lua

```diff
@@ -0,0 +1,363 @@
+-- Luagit2's Config Module Tests
+
+local fixer = require("Fixtures.fix_repo")
+local lfs = require("lfs")
+
+describe(" Config Methods Tests ", function()
+	local lib = require("luagit2")
+	local repo
+	local Config_ondisk , Config_Default, Config_System
+	local Config_Snapshot
+	local default_name , default_email
+	local sys_name , sys_email
+	local global_name, global_email
+	local current_directory_path = lfs.currentdir()
+
+	setup(function()
+		lib.luagit2_init()
+	end)
+
+	before_each(function()
+		fixer.set_repo("new_test_repo")
+		repo = lib.luagit2_repository_open("Fixtures/WORKON_REPO/.git")
+		Config_ondisk = lib.luagit2_config_open_ondisk(current_directory_path .. "/Fixtures/WORKON_REPO/.git/config")
+		Config_Default = lib.luagit2_config_open_default()
+		-- Config_System = lib.luagit2_config_open_system()
+
+		local name_handle = io.popen("git config --get user.name ")
+		default_name = name_handle:read("*a")
+		name_handle:close()
+
+		local email_handle = io.popen("git config --get user.email ")
+		default_email = email_handle:read("*a")
+		email_handle:close()
+
+		local sys_name_handle = io.popen("git config --system --get user.name ")
+		sys_name = sys_name_handle:read("*a")
+		sys_name_handle:close()
+
+		local sys_email_handle = io.popen("git config --system --get user.email ")
+		sys_email = sys_email_handle:read("*a")
+		sys_email_handle:close()
+
+		local global_name_handle = io.popen("git config --global --get user.name ")
+		global_name = global_name_handle:read("*a")
+		global_name_handle:close()
+
+		local global_email_handle = io.popen("git config --global --get user.email ")
+		global_email = global_email_handle:read("*a")
+		global_email_handle:close()
```

Comment:
It looks like there is a lot of repetition here. Maybe you could write a local function at the top of test file, just before the first describe that encapsulates opening a command, reading and closing it, then you just pass the command string here :)